frontend/src/pages/roles/components/UserRoleRow.tsx

```diff
@@ -0,0 +1,96 @@
+import Popover from '@mui/material/Popover';
+import RoleDTO from '@shared/dtos/RoleDTO';
+import UserDTO from '@shared/dtos/UserDTO';
+import { useRef, useState } from 'react';
+import FakeProfile from '../../../utility_components/FakeProfile/FakeProfile';
+
+export function UserRoleRow({
+    user,
+    roles,
+    removeRole,
+    addRole,
+}: {
+    user: UserDTO;
+    roles: RoleDTO[];
+    removeRole: (role: RoleDTO, userId: string) => void;
+    addRole: (role: RoleDTO, userId: string) => void;
+}) {
+    const [showRoles, setShowRoles] = useState<boolean>(false);
+    const ref = useRef(null);
+
+    function availableRolesToAdd() {
+        const rolesToAdd = roles.filter(
+            (role) => ['Admin', 'Default', ...user.roles.map((role) => role.name)].indexOf(role.name) === -1,
+        );
+
+        if (rolesToAdd.length === 0 && showRoles) {
+            setShowRoles(false);
+        }
+
+        return rolesToAdd;
+    }
+
+    return (
+        <>
+            <div className="flex flex-row gap-2 items-center w-full border-b-[1px] border-slate-300 pb-3">
+                <FakeProfile name={user.name} />
+                <div className="flex gap-1 flex-wrap w-full min-w-0">
+                    {user.roles.map((role) => {
+                        return (
+                            <span
+                                onClick={() => removeRole(role, user.id)}
+                                key={`${user.name} ${role.id} ${Math.random()}`}
```

Comment:
Let's remove `Math.random()`. We don't want unnecessary re-renders.

Reply:
done